controllers/auth_config_status_updater.go

```diff
@@ -41,43 +46,152 @@ func (u *AuthConfigStatusUpdater) Reconcile(ctx context.Context, req ctrl.Reques
 	} else {
 		// resource found and it is to be watched by this controller
 		// we need to update its status
-		if err := u.updateAuthConfigStatus(ctx, req.String(), &authConfig); err != nil {
-			logger.Info(err.Error())
+		if err := u.updateAuthConfigStatus(log.IntoContext(ctx, logger), req.String(), &authConfig); err != nil {
 			return ctrl.Result{Requeue: true}, nil
 		} else {
-			logger.Info("resource status updated")
 			return ctrl.Result{}, nil
 		}
 	}
 }
 
-func (u *AuthConfigStatusUpdater) updateAuthConfigStatus(ctx context.Context, cacheId string, authConfig *api.AuthConfig) error {
-	if len(u.Cache.FindKeys(cacheId)) != len(authConfig.Spec.Hosts) {
-		authConfig.Status.Ready = false
-		_ = u.Status().Update(ctx, authConfig)
-		return fmt.Errorf("resource not ready")
+func (u *AuthConfigStatusUpdater) updateAuthConfigStatus(ctx context.Context, cacheId string, authConfig *api.AuthConfig) (err error) {
+	logger := log.FromContext(ctx)
+
+	var condition api.Condition
+	linked, loose := u.partitionHostsByStatus(cacheId, authConfig)
+	if len(loose) > 0 {
+		condition = statusNotReady(api.StatusReasonHostNotLinked, "One or more hosts not linked to the resource")
+	} else {
+		condition = statusReady()
```

Comment:
As discussed before, maybe we could marked as `Ready` even when it's "partially" applied, some hosts were already taken.
And find a different status for the "hostsNotLinked"

Reply:
Yes, @didierofrivia. The option I made before for `Ready=false` when there are actually hosts linked to the AuthConfig was also bugging me. Maybe we can do better with more conditions instead of one boolean condition + reasons. Let us think about that for a while.